packages/circuits/circuits/sparseMerkleTree.circom

```diff
@@ -22,7 +24,12 @@ template SMTRootCalc(HEIGHT, ARITY) {
         // start doing checks
         if (i > 0) {
             // range check
-            path_indices[i] \ ARITY === 0;
+            mod[i] = Modulo();
+            mod[i].divisor <== ARITY;
+            mod[i].dividend <== path_indices[i];
+
+            path_indices_divided[i] <== mod[i].quotient;
```

Comment:
Why do we need the `path_indices_divided` variable? Here we need to check that `path_indices[i] < ARITY`.

Reply:
Maybe we can remove the sparseMerkleTree circuits?

Reply:
Yep that works as well, I don't think they're used anywhere else. We should do a bigger cleanup in a different PR.

Reply:
Oh, but the `SMTRootCalc` template is used in other components. It's not specific to an SMT, it works for any kind of merkle tree.